tests/FunctionalTests/Persistence/ResetPasswordRequestRepositoryTest.php

```diff
@@ -0,0 +1,216 @@
+    }
+
+    public function testPersistResetPasswordRequestPersistsRequestObject(): void
+    {
+        $fixture = new ResetPasswordRequestTestFixture();
+        $this->repository->persistResetPasswordRequest($fixture);
+
+        $result = $this->repository->findAll();
+
+        self::assertSame($fixture, $result[0]);
+    }
+
+    public function testGetUserIdentifierRetrievesObjectIdFromPersistence(): void
+    {
+        $fixture = new ResetPasswordRequestTestFixture();
+
+        $this->manager->persist($fixture);
+        $this->manager->flush();
+
+        $result = $this->repository->getUserIdentifier($fixture);
+
+        self::assertSame('1', $result);
+    }
+
+    public function testFindResetPasswordRequestReturnsObjectWithGivenSelector(): void
+    {
+        $fixture = new ResetPasswordRequestTestFixture();
+        $fixture->selector = '1234';
+
+        $this->manager->persist($fixture);
+        $this->manager->flush();
+
+        $result = $this->repository->findResetPasswordRequest('1234');
+
+        self::assertSame($fixture, $result);
+    }
+
+    public function testGetMostRecentNonExpiredRequestDateReturnsExpected(): void
+    {
+        $userFixture = new ResetPasswordUserTestFixture();
+        $this->manager->persist($userFixture);
+
+        $fixtureOld = new ResetPasswordRequestTestFixture();
+        $fixtureOld->requestedAt = new \DateTimeImmutable('-5 minutes');
+        $fixtureOld->user = $userFixture;
+        $this->manager->persist($fixtureOld);
+
+        $expectedTime = new \DateTimeImmutable();
+        $fixtureNewest = new ResetPasswordRequestTestFixture();
+        $fixtureNewest->expiresAt = new \DateTimeImmutable('+1 hours');
+        $fixtureNewest->requestedAt = $expectedTime;
+        $fixtureNewest->user = $userFixture;
+        $this->manager->persist($fixtureNewest);
+
+        $this->manager->flush();
+
+        $result = $this->repository->getMostRecentNonExpiredRequestDate($userFixture);
+
+        self::assertSame($expectedTime, $result);
+    }
+
+    public function testGetMostRecentNonExpiredRequestDateReturnsNullOnExpiredRequest(): void
+    {
+        $userFixture = new ResetPasswordUserTestFixture();
+        $this->manager->persist($userFixture);
+
+        $expiredFixture = new ResetPasswordRequestTestFixture();
+        $expiredFixture->user = $userFixture;
+        $expiredFixture->expiresAt = new \DateTimeImmutable('-1 hours');
+        $expiredFixture->requestedAt = new\DateTimeImmutable('-2 hours');
+        $this->manager->persist($expiredFixture);
+
+        $this->manager->flush();
+
+        $result = $this->repository->getMostRecentNonExpiredRequestDate($userFixture);
+
+        self::assertNull($result);
+    }
+
+    public function testGetMostRecentNonExpiredRequestDateReturnsNullIfRequestNotFound(): void
+    {
+        $userFixture = new ResetPasswordUserTestFixture();
```

Comment:
You're doing a great job prefixing these names so they don't pollute the user's auto-complete. But damn are these hard to understand due to it! What about `ResetPasswordTestFixtureUser` - the key part being that the *last* thing says `User` or `PasswordRequest` for the other entity. At least for me, that reads a lot better.

Reply:
+1 i concur
fixed and done